.github/workflows/continuous-integration.yml

```diff
@@ -82,7 +76,7 @@ jobs:
 
       - name: "Require symfony/messenger"
         run: "composer require --dev symfony/messenger --no-update"
-        if: "${{ startsWith(matrix.symfony-require, '4.') }}"
+        if: "${{ startsWith(matrix.symfony-require, '5.') }}"
```

Comment:
Why having a single job installing it ?

Reply:
Yeah this is something I wanted to investigate too. IIRC we had to install symfony/messenger for 4.x because it wasn't otherwise installed, while with 5.x it was. But I'm not 100% sure. But if it's true we can just remove this.

Reply:
… which would be strange because Symfony Messenger did exist in Symfony 4. I mean, I could just add the messenger to `require-dev` and drop this block. This would also allow to simplify a couple of tests and fix some Psalm errors.

Right now, we test that the bundle still works if Messenger is not installed. But then again, we don't do this for other Symfony components, only for Messenger. 🤷🏻‍♂️ 

I have no strong opinion here. Tell me which way to go and I'll change the code accordingly.

Reply:
I think it's good that we test both without and with messenger to ensure code works with both, but I don't have strong opinion. Btw this aproach was introduced in https://github.com/doctrine/DoctrineBundle/pull/817

Reply:
Okay, let's keep this as-is for now. We could revisit that in a follow-up. Maybe we could create a CI job that runs tests with minimal deps, so that we cover other optional deps as well.